code/__HELPERS/roundend.dm

```diff
@@ -335,6 +335,8 @@
 			else
 				parts += "<div class='panel greenborder'>"
 				parts += "<span class='greentext'>You managed to survive the events on [station_name()] as [M.real_name].</span>"
+				if(M.mind.assigned_role && (M.mind.assigned_role in GLOB.engineering_positions))
```

Comment:
```suggestion
				if(M.mind?.assigned_role in GLOB.engineering_positions)
```

M will always exists, mind needs a check and if assigned_role is null then the expression will simply return false

Reply:
Look at the greater code:
```cpp
	if(M.mind && !isnewplayer(M))
		if(M.stat != DEAD && !isbrain(M))
			if(EMERGENCY_ESCAPED_OR_ENDGAMED)
				if(!M.onCentCom() && !M.onSyndieBase())
					parts += "<div class='panel stationborder'>"
					parts += "<span class='marooned'>You managed to survive, but were marooned on [station_name()]...</span>"
				else
					parts += "<div class='panel greenborder'>"
					parts += "<span class='greentext'>You managed to survive the events on [station_name()] as [M.real_name].</span>"
			else
				parts += "<div class='panel greenborder'>"
				parts += "<span class='greentext'>You managed to survive the events on [station_name()] as [M.real_name].</span>"
				if(M.mind.assigned_role && (M.mind.assigned_role in GLOB.engineering_positions))
					SSachievements.unlock_achievement(/datum/achievement/engineering, C)
```
``M.mind`` has already been confirmed to exist within the if-block this code is running in, so it doesn't need a check here.

I had it bother to test if the assigned_role was set first mostly as a minor performance thing; I'm assuming just doing the quick test would be easier than having to iterate over GLOB.engineering_positions just to come to the same conclusion. It's a minor thing, but I don't believe it deserves a nitpick here.

Reply:
Well everyone or almost has an assigned_role, so it would be more of a performance fix to not have that check, the ? can be dropped tho, didnt see the mind check for some reason when looking for the M check

Reply:
*shrug
alright.